includes/class-wc-rest-authentication.php

```diff
@@ -38,6 +38,7 @@ class WC_REST_Authentication {
 	 * Initialize authentication actions.
 	 */
 	public function __construct() {
+		add_filter( 'determine_current_user', array( $this, 'authenticate_wccom' ), 14 );
```

Comment:
This should be moved into a new class, since one day we should stop support for all our current authenticate methods.

Reply:
Agree with Claudio, this functionality refers mainly to the helper and should reside alongside othe helper functionality. This also ensures that a helper update doesn't look past the code here.

Reply:
Moved in https://github.com/woocommerce/woocommerce/pull/24123/commits/0ffd4db6302a4d14e632ae0e365245b2c496ff67.